nova/virt/vmwareapi/vmops.py

```diff
@@ -2044,15 +2042,16 @@ def finish_migration(self, context, migration, instance, disk_info,
         self._update_instance_progress(context, instance,
                                        step=4,
                                        total_steps=RESIZE_TOTAL_STEPS)
-        # 5. Update ephemerals and attach volumes if necessary
+        # 5. Update ephemerals
         self._resize_create_ephemerals_and_swap(vm_ref, instance,
                                                 block_device_info)
-        if reattach_volumes:
-            self._attach_volumes(instance, block_device_info)
         self._update_instance_progress(context, instance,
                                        step=5,
                                        total_steps=RESIZE_TOTAL_STEPS)
-        # 6. Start VM
+
+        # 6. Attach the volumes (if necessary) and start the VM
+        if reattach_volumes:
+            self._attach_volumes(instance, block_device_info)
```

Comment:
Sorry, that's not what I meant. I wanted an extra 6th step for attaching, not moving the attachments into the power-on step. `RESIZE_TOTAL_STEPS = 7` instead of 6.